<a href="https://colab.research.google.com/github/Pranoti15/My-DS-Projects/blob/my-branch/Hypothesis_testing_Assignment_no_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Q1:**Hypothesis Formulation**:
- A company claims that their new energy drink increases focus and alertness.
Formulate the null and alternative hypotheses for testing this claim.

A:
**Null Hypothesis (H0)**: The new energy drink has no effect on focus and alertness. (i.e., there is no significant difference in focus and alertness between those who consume the drink and those who do not.)
**Alternative Hypothesis (H1)**: The new energy drink increases focus and alertness. (i.e., those who consume the drink have significantly higher focus and alertness compared to those who do not.)

This is a **one-tailed** hypothesis test, as we are specifically testing for an increase in focus and alertness.


Q2:**Significance Level Selection:**
- A researcher is conducting a study on the effects of exercise on weight loss. What
significance level should they choose for their hypothesis test and why?


A:1. **Hypothesis Formulation**

**Null Hypothesis (H0)**: Exercise has no significant effect on weight loss.
**Alternative Hypothesis (H1)**: Exercise has a significant effect on weight loss.

**Choosing the Significance Level (α)**

The significance level (α) represents the probability of rejecting the null hypothesis when it is actually true (Type I error). The choice of α depends on the context and consequences of errors:

**Common Significance Levels:**

**α=0.05 (5%)**: Standard choice in many scientific studies. Balances Type I and Type II errors.

**α=0.01 (1%)**: Used when making a Type I error is highly undesirable. Requires stronger evidence to reject H0

**α=0.10 (10%)**: Used in exploratory research where a more lenient threshold is acceptable.

2. **Recommended Significance Level:**

-If the study has moderate consequences of error and follows conventional research standards,
α=0.05 is appropriate.

-If incorrect conclusions could lead to serious medical or health-related consequences,
α=0.01 is preferable.

-If the study is exploratory and aims to detect potential trends without strict confirmatory claims,
α=0.10 might be considered.

3. **Interpretation**
If p-value < α: Reject H0, concluding that exercise has a statistically significant effect on weight loss.

If p-value ≥α: Fail to reject H0, suggesting insufficient evidence to conclude that exercise affects weight loss.

**Final Recommendation**
For most health and fitness studies, α=0.05 is the best choice as it ensures a reasonable balance between detecting a true effect and avoiding false positives. However, if medical implications are severe, α=0.01 should be considered.


Q3: **Interpreting p-values:**
- In a study investigating the effectiveness of a new teaching method, the calculated
p-value is 0.03. What does this p-value indicate about the null hypothesis?

A:
Interpreting the P-value in a Study Investigating a New Teaching Method
1. **Hypothesis Formulation**
Null Hypothesis (H0): The new teaching method has no effect on student performance compared to the traditional method.
Alternative Hypothesis (H1): The new teaching method significantly affects student performance compared to the traditional method.
2. **Understanding the P-value**
The p-value represents the probability of obtaining the observed results (or more extreme) under the assumption that the null hypothesis (H0) is true.
A p-value of 0.03 means that, assuming (H0) is true, there is a 3% chance of observing data as extreme as the study results due to random variation alone.
3.**Decision Rule**
Typically, a significance level (α) is set before analysis. Common thresholds are:
α=0.05 (5%): Conventional threshold for rejecting H0.
α=0.01 (1%): More stringent, used in cases requiring stronger evidence.
Comparison: Since the p-value (0.03) is less than α=0.05, we reject the null hypothesis at the 5% significance level.
4. **Interpretation of Results**
Statistical Conclusion: There is statistically significant evidence to suggest that the new teaching method affects student performance.
Practical Interpretation: While statistically significant, the real-world impact (effect size, practical significance) should also be considered before drawing strong conclusions.
5. **Limitations and Considerations**
Statistical Significance ≠ Practical Importance: A small p-value does not necessarily mean the effect is large or meaningful in real-world applications.
Possible Type I Error: Since we rejected H0, there is a 3% probability that we made a Type I error (incorrectly rejecting a true null hypothesis).
Study Design Factors: Sample size, effect size, and study biases should be evaluated to ensure robustness.
6. **Final Conclusion**
Given the p-value of 0.03, we conclude that the new teaching method has a statistically significant effect on student performance at the 5% level.
Further analysis (confidence intervals, effect size) is recommended to assess the practical implications of this finding.

Q4: **Type I and Type II Errors:**
- Describe a scenario in which a Type I error could occur in hypothesis testing. How
does it differ from a Type II error?

A:
**Type I and Type II Errors in Hypothesis Testing**
1. **Hypothesis Formulation**
Null Hypothesis (H0): The new teaching method has no effect on student performance.
Alternative Hypothesis (H1): The new teaching method has a significant effect on student performance.
2. **Definition of Type I and Type II Errors**
Type I Error (False Positive): Rejecting H0 when it is actually true.
Type II Error (False Negative): Failing to reject H0 when it is actually false.
3. **Type I Error (False Positive)**
Scenario:
A school tests a new teaching method and concludes that it improves student performance.
However, in reality, the new method has no real effect, and the observed improvement was due to chance or external factors.
Mathematical Explanation:
The probability of making a Type I error is denoted by α (significance level).
If α=0.05, there is a 5% chance of incorrectly rejecting H0.
This means that in 5 out of 100 experiments, the study might falsely conclude that the teaching method is effective when it is not.
Real-World Consequences:
The school adopts an ineffective teaching method, leading to wasted resources and potential harm to students' learning outcomes.
Researchers and policymakers make incorrect decisions based on flawed evidence.
4. **Type II Error (False Negative)**
Scenario:
A study investigates whether a new teaching method improves student performance but finds no significant effect and fails to reject H0.
However, in reality, the new teaching method actually improves learning outcomes, but the study failed to detect this effect.
Mathematical Explanation:
The probability of making a Type II error is denoted by β.
Power of a test is 1−β, representing the ability to detect a true effect.
A Type II error might occur due to:
-Small sample size

-High variability in student performance

-Inappropriate statistical test selection

Real-World Consequences:

-The school rejects an effective teaching method, missing an opportunity to enhance education.

-Educators continue using outdated methods that may not be as effective.

Q5: **Right-tailed Hypothesis Testing:**
- A manufacturer claims that their new light bulb lasts, on average, more than 1000
hours. Conduct a right-tailed hypothesis test with a significance level of 0.05, given a
sample mean of 1050 hours and a sample standard deviation of 50 hours.

A:

Step 1: **Formulating the Hypothesis**
The manufacturer claims that the new light bulb lasts more than 1000 hours on average. This suggests a right-tailed test.

Null Hypothesis (H0): The mean lifespan of the light bulb is 1000 hours.
H0:μ≤1000
Alternative Hypothesis (H1): The mean lifespan of the light bulb is greater than 1000 hours.
H1:μ>1000

Step 2: **Given Data**
Sample Mean = 1050 hours
Population Mean (μ0) = 1000 hours
Sample Standard Deviation (s) = 50 hours
Sample Size (n) = Not Given, so we assume a reasonable value for calculation (e.g., 30).
Significance Level (α) = 0.05

Step 3: **Compute the Test Statistic**
Since the population standard deviation is unknown, we use the t-test formula.
Let me calculate the t-score.

In [ ]:
import scipy.stats as stats
import numpy as np

# Given data
sample_mean = 1050
pop_mean = 1000
sample_std = 50
sample_size = 30

# Compute t-score
t_score = (sample_mean - pop_mean) / (sample_std / np.sqrt(sample_size))
t_score


5.477225575051661

**The computed t-score is 5.48.**

Step 4: **Determine the Critical Value**
Since this is a right-tailed test, we find the critical t-value at α=0.05 with (n-1) = 29 degrees of freedom.

Let me calculate the critical t-value.

In [ ]:
# Compute the critical t-value for a one-tailed test with alpha = 0.05
alpha = 0.05
df = sample_size - 1  # Degrees of freedom

critical_t_value = stats.t.ppf(1 - alpha, df)
critical_t_value

1.6991270265334972

The critical t-value for a right-tailed test at α=0.05 with 29 degrees of freedom is 1.70.

Step 5: **Compare t-score with Critical Value**
t-score = 5.48
Critical t-value = 1.70
Since 5.48 > 1.70, we reject the null hypothesis.

Step 6: **Interpretation**
At a 5% significance level, there is strong statistical evidence to support the manufacturer's claim that the new light bulb lasts, on average, more than 1000 hours.

Q6: **Two-Tailed Hypothesis Testing:**
- A researcher wants to determine if there is a difference in mean exam scores between
two groups of students. Formulate the null and alternative hypotheses for this study as a
two-tailed test.

A:
**Two-Tailed Hypothesis Testing**
Step 1: **Formulating the Hypothesis**
The researcher wants to test whether there is a difference in mean exam scores between two groups of students. This suggests a two-tailed test.

Null Hypothesis (H0): There is no difference in the mean exam scores between the two groups.
H0:μ 1=μ 2
(The means of both groups are equal.)

Alternative Hypothesis (H1): There is a difference in the mean exam scores between the two groups.
H1:μ1≠μ2
(The means are not equal, meaning one group has a significantly higher or lower mean score than the other.)

Step 2: **Given Data**
Let’s assume the following data for calculation purposes:

Group 1: Sample Mean, Sample Standard Deviation (s1), Sample Size (n1)
Group 2: Sample Mean, Sample Standard Deviation (s2), Sample Size (n2)

Significance Level: α=0.05

Since the population standard deviations are unknown, we will use the independent two-sample t-test.

Step 3: **Compute the Test Statistic**
The test statistic for an independent two-sample t-test is given by:
Let me perform this calculation with assumed values:


Group 1:
x1=78,
s1=10,
n1=30

Group 2:
x2=72,
s2=12,
n2=30

In [ ]:
# Re-import necessary libraries since the execution state was reset
import scipy.stats as stats
import numpy as np

# Given data
x1, s1, n1 = 78, 10, 30  # Group 1: mean, std dev, sample size
x2, s2, n2 = 72, 12, 30  # Group 2: mean, std dev, sample size

# Compute t-score for independent two-sample t-test
t_statistic = (x1 - x2) / np.sqrt((s1**2 / n1) + (s2**2 / n2))
t_statistic


2.1038606199548298

**The computed t-score is 2.10.**

Step 4: **Determine the Critical Value**
Since this is a two-tailed test, we find the critical t-value at α=0.05 with
𝑑𝑓=𝑛1+𝑛2−2
  =30+30−2
  =58 degrees of freedom.

Let me calculate the critical t-value.

In [ ]:
# Compute the critical t-value for a two-tailed test with alpha = 0.05
alpha = 0.05
df = n1 + n2 - 2  # Degrees of freedom

# Critical t-value for two-tailed test
critical_t_value = stats.t.ppf(1 - alpha / 2, df)
critical_t_value


2.0017174841452356

**The critical t-value for a two-tailed test at α=0.05 with 58 degrees of freedom is 2.00.**

Step 5: **Compare t-score with Critical Value**
t-score = 2.10
Critical t-value = 2.00
Since 2.10 > 2.00, we reject the null hypothesis.

Step 6: **Interpretation**
At a 5% significance level, there is statistically significant evidence to conclude that there is a difference in the mean exam scores between the two student groups.

Q7: **One-sample t-test:**
- A manufacturer claims that the mean weight of their cereal boxes is 500 grams. A
sample of 30 cereal boxes has a mean weight of 490 grams and a standard deviation of
20 grams. Conduct a one-sample t-test to determine if there is evidence to support the
manufacturer's claim at a significance level of 0.05.

A:
**One-Sample t-Test**
Step 1: **Formulating the Hypothesis**
The manufacturer claims that the mean weight of cereal boxes is 500 grams. We will test whether the sample provides evidence to refute this claim.

Null Hypothesis (H0): The mean weight of the cereal boxes is 500 grams.
H0:μ=500
Alternative Hypothesis (H1): The mean weight is not 500 grams (two-tailed test).
H1:μ≠500
Step 2: **Given Data**
Sample Mean = 490 grams
Population Mean (μ0) = 500 grams
Sample Standard Deviation (s) = 20 grams
Sample Size (n) = 30
Significance Level (α) = 0.05
Step 3: **Compute the Test Statistic**
Since the population standard deviation is unknown, we use the one-sample t-test formula:

In [1]:
# Re-import necessary libraries since the execution state was reset
import scipy.stats as stats
import numpy as np

# Given data
sample_mean = 490
pop_mean = 500
sample_std = 20
sample_size = 30

# Compute t-score
t_score = (sample_mean - pop_mean) / (sample_std / np.sqrt(sample_size))
t_score


-2.7386127875258306

**The computed t-score is -2.74.**

Step 4: **Determine the Critical Value**
Since this is a two-tailed test, we find the critical t-value at α=0.05 with
df=n−1=30−1=29 degrees of freedom.

Let me calculate the critical t-value.

In [2]:
# Compute the critical t-value for a two-tailed test with alpha = 0.05
alpha = 0.05
df = sample_size - 1  # Degrees of freedom

# Critical t-value for two-tailed test
critical_t_value = stats.t.ppf(1 - alpha / 2, df)
critical_t_value


2.045229642132703

The critical t-value for a two-tailed test at α=0.05 with 29 degrees of freedom is ±2.05.

Step 5: **Compare t-score with Critical Value**
t-score = -2.74
Critical t-value = ±2.05
Since −2.74<−2.05, we reject the null hypothesis.

Step 6: **Interpretation**
At a 5% significance level, there is statistically significant evidence to conclude that the mean weight of the cereal boxes differs from 500 grams. This suggests that the manufacturer's claim may not be accurate.

Q8:
 **Two-sample t-test:**
- A researcher wants to compare the mean reaction times of two different groups of
participants in a driving simulation study. Group A has a mean reaction time of 0.6
seconds with a standard deviation of 0.1 seconds, while Group B has a mean reaction
time of 0.55 seconds with a standard deviation of 0.08 seconds. Conduct a two-sample
t-test to determine if there is a significant difference in mean reaction times between the
groups at a significance level of 0.01.

A:
**Two-Sample t-Test**
Step 1: **Formulating the Hypothesis**
The researcher wants to test whether there is a significant difference in mean reaction times between two groups. This is a two-tailed test.

Null Hypothesis (H0): There is no difference in mean reaction times between Group A and Group B.
H0:μA=μB (The means of both groups are equal.)

Alternative Hypothesis (H1): There is a difference in mean reaction times between Group A and Group B.
HA:μA≠μB (The means are not equal, meaning one group has significantly higher or lower reaction times than the other.)

Step 2: **Given Data**
Group A:

Sample Mean (xA) = 0.6 seconds
Sample Standard Deviation (sA) = 0.1 seconds
Sample Size (nA) = 30 (assumed)
Group B:

Sample Mean (xB) = 0.55 seconds
Sample Standard Deviation (sB) = 0.08 seconds
Sample Size (nB) = 30 (assumed)
Significance Level (α): 0.01

Step 3: **Compute the Test Statistic**
Since the population standard deviations are unknown, we use the independent two-sample t-test formula:

Let me calculate the t-score.

In [3]:
# Given data
x_A, s_A, n_A = 0.6, 0.1, 30  # Group A: mean, std dev, sample size
x_B, s_B, n_B = 0.55, 0.08, 30  # Group B: mean, std dev, sample size

# Compute t-score for independent two-sample t-test
t_statistic = (x_A - x_B) / np.sqrt((s_A**2 / n_A) + (s_B**2 / n_B))
t_statistic

2.138497306920751

**The computed t-score is 2.14.**

Step 4: Determine the Critical Value
Since this is a two-tailed test, we find the critical t-value at α=0.01 with

df=nA+nB−2=30+30−2=58 degrees of freedom.

Let me calculate the critical t-value.

In [4]:
# Compute the critical t-value for a two-tailed test with alpha = 0.01
alpha = 0.01
df = n_A + n_B - 2  # Degrees of freedom

# Critical t-value for two-tailed test
critical_t_value = stats.t.ppf(1 - alpha / 2, df)
critical_t_value


2.663286953537658

**The critical t-value for a two-tailed test at α=0.01 with 58 degrees of freedom is ±2.66.**

Step 5: **Compare t-score with Critical Value**

t-score = 2.14,
Critical t-value = ±2.66

Since -2.66< 2.14 < 2.66, we fail to reject the null hypothesis.

Step 6: **Interpretation**
At a 1% significance level, there is not enough statistical evidence to conclude that the mean reaction times of Group A and Group B are significantly different. This suggests that any observed difference in reaction times could be due to chance rather than a true difference.

Q9:
**Process Control Example:**
- A call center manager implements a new training program aimed at reducing call
waiting times. The average waiting time before the training program was 4.5 minutes, and
after the program, it is measured to be 4.0 minutes with a standard deviation of 0.8
minutes. Conduct a hypothesis test to determine if there is evidence that the training
program has reduced waiting times, using a significance level of 0.05.

A:
**Process Control Hypothesis Testing: Effect of Training Program on Call Waiting Times**

Step 1: **Formulating the Hypothesis**
The call center manager wants to determine whether the new training program has reduced call waiting times. This suggests a left-tailed (one-tailed) test.

Null Hypothesis (H0): The training program has no effect or has not reduced the average waiting time.

H0:μ≥4.5

Alternative Hypothesis (H1): The training program has reduced the average waiting time.

H1:μ < 4.5

Step 2: **Given Data**
Before Training: Population Mean (μ0) = 4.5 minutes
After Training: Sample Mean (x) = 4.0 minutes
Sample Standard Deviation (s) = 0.8 minutes
Sample Size (n) = 30 (assumed)
Significance Level (a) = 0.05

Step 3: **Compute the Test Statistic**
Since the population standard deviation is unknown, we use the one-sample t-test formula:

Let me calculate the t-score.

In [5]:
# Given data
sample_mean = 4.0
pop_mean = 4.5
sample_std = 0.8
sample_size = 30

# Compute t-score
t_score = (sample_mean - pop_mean) / (sample_std / np.sqrt(sample_size))
t_score


-3.4232659844072884

**The computed t-score is -3.42.**

Step 4: **Determine the Critical Value**
Since this is a left-tailed test, we find the critical t-value at α=0.05 with

df=n−1=30−1=29 degrees of freedom.

Let me calculate the critical t-value.

In [6]:
# Compute the critical t-value for a left-tailed test with alpha = 0.05
alpha = 0.05
df = sample_size - 1  # Degrees of freedom

# Critical t-value for one-tailed test
critical_t_value = stats.t.ppf(alpha, df)
critical_t_value


-1.6991270265334977

**The critical t-value for a left-tailed test at α=0.05 with 29 degrees of freedom is -1.70.**

Step 5: **Compare t-score with Critical Value**

t-score = -3.42,
Critical t-value = -1.70

Since -3.42<-1.70, we reject the null hypothesis.

Step 6: **Interpretation**

At a 5% significance level, there is statistically significant evidence to conclude that the new training program has successfully reduced call waiting times.

Q10:
**Interpreting Results:**
- After conducting a hypothesis test, the calculated p-value is 0.02. What can you
conclude about the null hypothesis based on this result, assuming a significance level of 0.05?

A:
**Interpreting Hypothesis Test Results Using the p-Value Approach**

Step 1: **Understanding the Hypothesis Setup**
In hypothesis testing, we compare the p-value to the significance level (a) to determine whether to reject the null hypothesis.

Null Hypothesis (H0): The claim we assume to be true before testing.
Alternative Hypothesis (H1): The claim we want to test.

Step 2: **Given Information**
Calculated p-value = 0.02,
Significance Level (a) = 0.05

Step 3: **Decision Rule**
If p≤α, reject the null hypothesis (H0).

If p>α, fail to reject the null hypothesis.

Since 0.02 < 0.05, we reject the null hypothesis.

Step 4: **Interpretation**

At a 5% significance level, there is sufficient statistical evidence to reject the null hypothesis. This means the test results support the alternative hypothesis, suggesting that the effect or difference observed in the sample is statistically significant.